File: hto/57-1/hto_57-1.java
class Solution {
    public int[][] findContinuousSequence(int target) {
        List<int[]> list = new ArrayList<>();
        int t = target / 2 + 1;
        for (int l = 1, r = 1, sum = 0; r <= t; r++) {
            sum += r;
            while (sum > target) {
                sum -= l++;
            }
            if (sum == target) {
                int[] temp = new int[r - l + 1];
                for (int i = 0; i < temp.length; i++) {
                    temp[i] = l + i;
                }
                list.add(temp);
            }
        }

        return list.toArray(new int[0][]);
    }
}

File: hto/57-1/hto_57-1.cpp
class Solution {
public:
    vector<vector<int>> findContinuousSequence(int target) {
        vector<vector<int>> res;
        vector<int> tmp;
        int i = 1; 
        int sum = 3;
        int mid = target / 2 + 1;
        for (int j = 3; j <= mid; j++) {
            sum += j;
            while (sum > target) {
                sum -= i++;
            }
            if (sum == target) {
                tmp.clear();
                for (int k = i; k <= j; k++) {
                    tmp.push_back(k);
                }
                res.push_back(tmp);
            }
        }
        return res;
    }
};